### ISSUE TITLE
bee oom loop even with 6gb ram

### ISSUE THREAD

**Issue by darkobas2** [ethersphere/bee#3123]
⚠️ Support requests in an issue-format will be closed immediately. For support, go to Swarm's [Discord](https://discord.gg/XGNBEKktkD).

### Context
1.7.0

### Summary
Bee uses 6gb+ memory

### Expected behavior
Should work fine with 3-4gb ?!

### Actual behavior
````
    Last State:     Terminated
      Reason:       OOMKilled
      Exit Code:    137
      Started:      Tue, 26 Jul 2022 18:01:30 +0200
      Finished:     Tue, 26 Jul 2022 18:17:54 +0200
````
![bee-oom](https://user-images.githubusercontent.com/93942788/181058979-53bea4ca-b4b0-4bbb-aece-0132bed903ea.png)

### Steps to reproduce
N/A

### Possible solution
nuke db

logs:
[bee.log](https://github.com/ethersphere/bee/files/9191639/bee.log)
[bee-previous.log](https://github.com/ethersphere/bee/files/9191640/bee-previous.log)
[dump1.log](https://github.com/ethersphere/bee/files/9191643/dump1.log)
[dump2.log](https://github.com/ethersphere/bee/files/9191644/dump2.log)

[heap.log](https://github.com/ethersphere/bee/files/9196266/heap.log)
[allocs.log](https://github.com/ethersphere/bee/files/9196267/allocs.log)

localstore log:
[LOG.log](https://github.com/ethersphere/bee/files/9191659/LOG.log)
[LOG.old.log](https://github.com/ethersphere/bee/files/9191660/LOG.old.log)

**Comment by darkobas2** [comment]
![profile004](https://user-images.githubusercontent.com/93942788/187875278-31055182-0402-4284-b41d-4e98f2aac7a9.png)
![profile002](https://user-images.githubusercontent.com/93942788/187875283-32316094-4d09-4a91-baa7-b56cbf44cabb.png)
![profile001](https://user-images.githubusercontent.com/93942788/187875287-7a42b36c-937c-42ea-92a8-693791fdb641.png)


some more dumps from today

**Comment by aloknerurkar** [comment]
@darkobas2 Thanks for all the details! From the profiles it looks like this is a new bug. The UnreserveBatch doesnt have any limit on how many chunks it will remove. Seems like there are some pretty huge batch evictions. This would cause this batch to grow very big during GC and run out of memory.

**Comment by mrekucci** [comment]
@darkobas2 can you try the following fix: #3247

**Comment by aloknerurkar** [comment]
> @darkobas2 can you try the following fix: #3247

@mrekucci I created an image for him on top of 1.7.0 and it has been running since Saturday without restarts.

**Comment by mrekucci** [comment]
> > @darkobas2 can you try the following fix: #3247
> 
> @mrekucci I created an image for him on top of 1.7.0 and it has been running since Saturday without restarts.
Oh, nice! Didn't know that. All good then 🙏
